tests/testrom.c

```diff
@@ -243,6 +246,10 @@ int main() {
 		fflush(stdout);
 		debugf("**** Starting test: %s\n", tests[i].name);
 
+		// Do a complete cache flush before running each test
+		data_cache_hit_writeback_invalidate(MEM_START, MEM_LENGTH);
+		inst_cache_hit_invalidate(MEM_START, MEM_LENGTH);
```

Comment:
I think it would be better to abstract this in proper functions: data_cache_writeback_invalidate_all() and inst_cache_invalidate_all().

The best implementation would be to use the "cache by index" family of cache opcodes that manipulates the cache using the internal index rather than the memory address. Then you iterate the index on all the available slots (need to check how many cache slots are in VR4300) so that you fish everything with maybe 64 or 128 iterations rather than several thousands like in this implementation.

If you don't want to research this, it's to add the new API for now with this implementation and leave a TODO for later.

Reply:
Done without index ops for now.